src/components/ColorPicker/index.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { useEffect, useRef, useState } from "react";
+import { ActionCreator } from "redux";
+import { Popover, Tooltip } from "antd";
+import { HexColorInput, HexColorPicker } from "react-colorful";
+import classNames from "classnames";
+import { useDebounce } from "use-debounce";
+import { ColorChange } from "@aics/simularium-viewer";
+
+import { AGENT_COLORS } from "../../containers/ViewerPanel/constants";
+import {
+    SetColorChangeAction,
+    SetRecentColorsAction,
+} from "../../state/selection/types";
+
+import styles from "./style.css";
+
+interface ColorPickerProps {
+    childrenHaveDifferentColors?: boolean;
+    selectedColor: string;
+    agentName: string;
+    tags: string[];
+    recentColors: string[];
+    setColorChange: ActionCreator<SetColorChangeAction>;
+    setRecentColors: ActionCreator<SetRecentColorsAction>;
+}
+
+const ColorPicker = ({
+    agentName,
+    tags,
+    recentColors,
+    setColorChange,
+    setRecentColors,
+    selectedColor,
+    childrenHaveDifferentColors,
+}: ColorPickerProps) => {
+    const [color, setColor] = useState(selectedColor);
+    const [debouncedColor] = useDebounce(color, 250);
+    const isInitialRender = useRef(true);
```

Comment:
Remove? It doesn't seem to be in use anymore

Reply:
it is used, but it's accessed as isInitialRender.current

Reply:
Oh I see, I skipped over how it was being used in the `if` check below. My bad!